pkg/app/ops/planpreviewoutputcleaner/cleaner.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package planpreviewoutputcleaner
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipe/pkg/filestore"
+)
+
+const (
+	outputTTL = 48 * time.Hour
+	interval  = 24 * time.Hour
+	prefix    = "command-output/"
+)
+
+type Cleaner struct {
+	lister  filestore.Lister
+	deleter filestore.Deleter
+	logger  *zap.Logger
+}
+
+func NewCleaner(l filestore.Lister, d filestore.Deleter, logger *zap.Logger) *Cleaner {
+	return &Cleaner{
+		lister:  l,
+		deleter: d,
+		logger:  logger.Named("planpreview-output-cleaner"),
+	}
+}
+
+func (c *Cleaner) Run(ctx context.Context) error {
+	c.logger.Info("start running planpreview ouput cleaner")
+
+	t := time.NewTicker(interval)
+	for {
+		select {
+		case <-ctx.Done():
+			break
+
+		case t := <-t.C:
+			c.clean(ctx, t)
+		}
+	}
+
+	c.logger.Info("planpreview output cleaner has been stopped")
+	return nil
+}
+
+func (c *Cleaner) clean(ctx context.Context, now time.Time) error {
+	objects, err := c.lister.List(ctx, prefix)
+	if err != nil {
+		c.logger.Error("failed to list planpreview output objects",
+			zap.String("prefix", prefix),
+			zap.Error(err),
+		)
+		return err
+	}
+
+	ttl := outputTTL.Seconds()
+	deletes := 0
+
+	for _, obj := range objects {
+		if float64(now.Unix()-obj.UpdatedAt) <= ttl {
```

Comment:
```suggestion
		if time.Since(time.Unix(obj.UpdatedAt, 0)) <= ttl {
```
nits

Reply:
Since `now` is passed to `clean` function only one time and be used for all objects in this loop so let me keep this as it is.

Reply:
ah I see, get your point 👍 in that case instead of using now passed by cleaner caller, how about initializing `now` before this for loop, just want to keep unified with other cleaners (all not pass `now` from caller but init that in clear function. Of cost it's a nit so up to you 🙆‍♀️

Reply:
Sure. There is no objection to it. 👍